test-data/unit/check-selftype.test

```diff
@@ -1654,7 +1654,7 @@ class C:
     def bar(self) -> Self: ...
     foo: Callable[[S, Self], Tuple[Self, S]]
 
-reveal_type(C().foo)  # N: Revealed type is "def [S] (S`-1, __main__.C) -> Tuple[__main__.C, S`-1]"
+reveal_type(C().foo)  # N: Revealed type is "def [S] (S`1, __main__.C) -> Tuple[__main__.C, S`1]"
```

Comment:
Not sure if I remember the exact rules, but shouldn't S continue to use a negative ID, since it's a method-scoped typevar instead of a class-scoped one?

Reply:
Yes, but we didn't follow this convention since when we started using `freshen_function_type_vars()` in `checkmember` ~5 years ago. This is because unification variables have positive ids. But yeah, at some point we should try to clean this up.